### ISSUE TITLE
All metrics evaluate to 0.0 when using  tf.contrib.estimator.InMemoryEvaluatorHook

### ISSUE THREAD

**Issue by tavramov** [tensorflow/tensorflow#21590]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Slight modification to cnn_mnist.py to include `InMemoryEvaluatorHook` . Original code can be found here [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/layers/cnn_mnist.py)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:N/A
- **TensorFlow installed from (source or binary)**: binary (i think)
- **TensorFlow version (use command below)**:1.9.0 (with PR#20822)
- **Python version**: 3.5.2
- **Bazel version (if compiling from source)**:N/A
- **GCC/Compiler version (if compiling from source)**:5.4.0
- **CUDA/cuDNN version**:7.5
- **GPU model and memory**: 2x TITAN X (12189MiB) & Quadro P5000 (16273MiB)
- **Exact command to reproduce**:  python3 cnn_mnist.py

### Describe the problem
When evaluating during training with `tf.contrib.estimator.InMemoryEvaluatorHook`, all evaluation metrics are evaluated to 0.0 after the very first time the evaluation hook is triggered. My suspicion is that it is caused by the use of `tf.estimator.inputs.numpy_input_fn`.

### Source code:
Attached is the script I used. Basically, just added the following line:
` evaluator = tf.contrib.estimator.InMemoryEvaluatorHook(estimator=mnist_classifier, input_fn=eval_input_fn, every_n_iter=50)` and passed the hook to `mnist_classifer.train()`

[cnn_mnist.txt](https://github.com/tensorflow/tensorflow/files/2285277/cnn_mnist.txt)

Here is the relevant output from running the training for 200 steps:

INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:35
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:37
INFO:tensorflow:Saving dict for global step 0: accuracy = 0.0915, global_step = 0, loss = 2.30028
INFO:tensorflow:loss = 2.29529, step = 0
...
INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:37
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:37
INFO:tensorflow:Saving dict for global step 50: accuracy = 0.0, global_step = 50, loss = 0.0
...
INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:38
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:38
INFO:tensorflow:Saving dict for global step 100: accuracy = 0.0, global_step = 100, loss = 0.0
INFO:tensorflow:global_step/sec: 103.013
INFO:tensorflow:loss = 2.27342, step = 100 (0.971 sec)
...
INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:38
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:38
INFO:tensorflow:Saving dict for global step 150: accuracy = 0.0, global_step = 150, loss = 0.0
...
INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:39
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:39
INFO:tensorflow:Saving dict for global step 200: accuracy = 0.0, global_step = 200, loss = 0.0
...
INFO:tensorflow:Saving checkpoints for 200 into ./tmp/mnist_convnet_model/model.ckpt.
INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:39
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:39
INFO:tensorflow:Saving dict for global step 200: accuracy = 0.0, global_step = 200, loss = 0.0
INFO:tensorflow:Loss for final step: 2.25765.
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
...
INFO:tensorflow:Starting evaluation at 2018-08-14-02:15:40
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from ./tmp/mnist_convnet_model/model.ckpt-200
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-08-14-02:15:40
INFO:tensorflow:Saving dict for global step 200: accuracy = 0.2611, global_step = 200, loss = 2.25732
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 200: ./tmp/mnist_convnet_model/model.ckpt-200
{'global_step': 200, 'loss': 2.2573187, 'accuracy': 0.26109999}

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ispirmustafa: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ispirmustafa: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @karmel: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
We currently don't have anybody working on this. It would be great if you could help us by working on this and submitting a PR. Let us know if you need further clarification. Thanks!

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by rthadur** [comment]
Contrib has been depreciated here in Tensorflow repo and moved to tensorflow/addons , please reopen the issue in https://github.com/tensorflow/addons/issues if this exists in latest version. Thank you
